### ISSUE TITLE
 An error has occurred during metrics collection:500 internal server err

### ISSUE THREAD

**Issue by NotSharath** [prometheus-net/prometheus-net#89]
https://github.com/prometheus/pushgateway/issues/191 #

When I am trying to use Pushgateway (c#) to publish metrics, I get below err

> 

2 error(s) occurred:
* gathered metric family process_start_time_seconds has help "**Start time of the process since unix epoch in seconds**" but should have "**Start time of the process since unix epoch in seconds.**"
* gathered metric family process_cpu_seconds_total has help "**Total user and system CPU time spent in seconds**" but should have "**Total user and system CPU time spent in seconds.**"

> 

In my prometheus localhost:9090 dashboard I see the status as 500 intrnl err

http://localhost:9091/metrics | DOWN | instance="localhost:9091" | 581ms ago | server returned HTTP status 500 Internal Server Error
 
-- I cloned and checked in the code (**DotNerStatsCollector.cs**) class I saw

            _startTime = metrics.CreateGauge("process_start_time_seconds", "Start time of the process since unix epoch in seconds");
            _cpuTotal = metrics.CreateCounter("process_cpu_seconds_total", "Total user and system CPU time spent in seconds");

here the arguments dont have period in the end !!!!!
So, is period the culprit ??? Am I running uncompatable versions ? is it a bad unit test ???

how to pass through this issue.

**Comment by SKRVulupala** [comment]
@sandersaares I did clone the code to my local and added the periods to the code- than the metrics were successfully published
```
_startTime = metrics.CreateGauge("process_start_time_seconds", "Start time of the process since unix epoch in seconds.");
        _cpuTotal = metrics.CreateCounter("process_cpu_seconds_total", "Total user and system CPU time spent in seconds.");
```
do you want me to submit  a PR for this or can you kindly fix the test case validation, so that I can get past this blocker. 

- Respect

**Comment by tuliogomesbarbosa** [comment]
@NotSharath please submit a PR! I'm facing the same problem.

**Comment by sandersaares** [comment]
Thank you for the report! I see what is happening here:

1. pushgateway requires all metrics with the same name to have the same help text (at least in certain situations).
1. However, if you have the same metric reported from different sources (prometheus-net and something else) you cannot assume that the help text 100% matches.
1. This violated assumption causes an error.

This is basically a constraint imposed by the current pushgateway design, as there is no reason to assume that such data is globally synchronized. There also exist other Prometheus libraries that export metrics named the same, with even more different descriptions than just a difference of periods at the end.

It appears that Pushgateway maintainers are considering a fix for this in https://github.com/prometheus/pushgateway/issues/194. Please upvote that issue if this is important to you.

For prometheus-net I will synchronize the descriptions with the Go implementation, which does include the two periods you mention. This change will shortly be available in a prerelease version.

**Comment by sandersaares** [comment]
This fix will be live as 2.1.2 in approximately 1 hour o.
